atlp-rwanda/e-comm-team-brogrammers-bn
# #184581335 Add Multi-Factor Authentication(MFA)

#### What does this PR do?
This is to add multi-factor authentication to increase security the login process, for users that turn on MFA.
#### Description of Task to be completed?
GIVEN there is a request to login a user to the website
WHEN 2FA is enabled on the particular user
THEN Further request for a 2FA code is requested and once the code is received and validated, the user is allowed to login(via a jwt token).
#### How should this be manually tested?

- run `npm install`
- run `npm run migrate:all`
- run `npm run start`
- Login a new user
- Enable MFA with `POST` request at `/users/enable-mfa` providing a valid token for auth
- Try logging in again, now you shouldn't get a token instead message to check your inbox
- Copy the the code and sent it with `POST` to `/users/verify-mfa` you should get a jwt token

#### Any background context you want to provide?
- To disable MFA a `POST` request can be sent to `/users/disable-mfa` with valid jwt token
#### What are the relevant pivotal tracker/Trello stories?
[https://www.pivotaltracker.com/n/projects/2632830/stories/184581335](url)
#### Screenshots (if appropriate)
<img width="1012" alt="Screenshot 2023-03-14 at 16 53 45" src="https://user-images.githubusercontent.com/57767759/225052641-86b0e678-8f8f-4f7f-b6bb-a999a8e826c1.png">
<img width="1256" alt="Screenshot 2023-03-14 at 17 22 08" src="https://user-images.githubusercontent.com/57767759/225052686-faaf1a65-1c21-4e30-8a49-70c8f68bf6c6.png">

#### Questions:
